components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java

```diff
@@ -1102,18 +1113,21 @@ public API getAPIByContextAndVersion(String context, String version, String depl
                         api.setApiType(apiType);
                         api.setPolicy(resultSet.getString("API_TIER"));
                         api.setContext(resultSet.getString("CONTEXT"));
-                        String revision = resultSet.getString("REVISION_UUID");
                         api.setStatus(resultSet.getString("STATUS"));
+                        String revision = resultSet.getString("REVISION_UUID");
                         api.setIsDefaultVersion(isAPIDefaultVersion(connection, provider, name, version));
-                        api.setPolicy(getAPILevelTier(connection, apiUuid, revision));
-                        if (deployment.equals(deploymentName)) {
-                            if (APIConstants.API_PRODUCT.equals(apiType)) {
-                                attachURlMappingDetailsOfApiProduct(connection, api);
-                                return api;
+                        if (isExpand) {
+                            api.setPolicy(getAPILevelTier(connection, apiUuid, revision));
+                            if (deployment.equals(deploymentName)) {
+                                if (APIConstants.API_PRODUCT.equals(apiType)) {
+                                    attachURlMappingDetailsOfApiProduct(connection, api);
+                                }
+                                attachURLMappingDetails(connection, revision, api);
                             }
-                            attachURLMappingDetails(connection, revision, api);
-                            return api;
+                        } else {
+                            api.setPolicy(null);
                         }
+                        return api;
```

Comment:
Looks like the previous logic was there to return the API only if the API's deployment (from DB) matches with the given `deployment` from method are equal and it seems to be slightly changed now. 

Should this be something like

```
if (deployment.equals(deploymentName)) {
    if (isExpand) {
       // expanding logic
    }
    return api;
}
return null;
```